src/Storage/Device/Local.php

```diff
@@ -69,24 +71,70 @@ public function getPath($filename): string
      *
      * @param string $source
      * @param string $path
+     * @param int $chunk
+     * @param int $chunks
+     * @param string tmp
      *
      * @throws \Exception
      *
-     * @return bool
+     * @return int
      */
-    public function upload($source, $path): bool
+    public function upload($source, $path, $chunk = 1, $chunks = 1): int
     {
         if (!\file_exists(\dirname($path))) { // Checks if directory path to file exists
             if (!@\mkdir(\dirname($path), 0755, true)) {
                 throw new Exception('Can\'t create directory: ' . \dirname($path));
             }
         }
 
-        if (\move_uploaded_file($source, $path)) {
-            return true;
-        }
+        //move_uploaded_file() verifies the file is not tampered with
+        if($chunks == 1) {
+            if (!\move_uploaded_file($source, $path)) {
+                throw new Exception('Can\'t upload file ' . $path);
+            }
+            return $chunks;
+        } else {
+            $tmp = \dirname($path) . "/tmp/chunks.log";
+
+            if (!\file_exists(\dirname($tmp))) { // Checks if directory path to file exists
+                if (!@\mkdir(\dirname($tmp), 0755, true)) {
+                    throw new Exception('Can\'t create directory: ' . \dirname($tmp));
+                }
+            }
+            if(!file_put_contents($tmp, "{$chunk}\n", FILE_APPEND)) {
+                throw new Exception('Can\'t write chunk log ' . $tmp);
+            }
 
-        return false;
+            $chunkLogs = file($tmp);
+            if(!$chunkLogs) {
+                throw new Exception("Unable to read chunk log " . $tmp);
```

Comment:
```suggestion
                throw new Exception('Unable to read chunk log ' . $tmp);
```

lets try to be consistent with the single quotes 👍🏻